File: Demo/Sources/ViewControllers/Notifications/NotificationsViewController.m
//
//  Globus
//

#import "NotificationsViewController.h"
#import "NotificationTopDecorView.h"
#import "NotificationBottomDecorView.h"
#import "NotificationView.h"

@interface NotificationsViewController ()

@property(nonatomic, weak) IBOutlet UISegmentedControl* segmentedControl;
@property(nonatomic, weak) IBOutlet UISwitch* useTopDecorView;
@property(nonatomic, weak) IBOutlet UISwitch* useBottomDecorView;;
@property(nonatomic, weak) IBOutlet UITextField* textField;

@end

@implementation NotificationsViewController

#pragma mark - UIViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [GLBNotificationManager setStatusBarStyle:UIStatusBarStyleLightContent];
    
    [_segmentedControl removeAllSegments];
    [_segmentedControl insertSegmentWithTitle:@"List" atIndex:_segmentedControl.numberOfSegments animated:NO];
    [_segmentedControl insertSegmentWithTitle:@"Stack" atIndex:_segmentedControl.numberOfSegments animated:NO];
    switch ([GLBNotificationManager displayType]) {
        case GLBNotificationManagerDisplayTypeList:
            _segmentedControl.selectedSegmentIndex = 0;
            break;
        case GLBNotificationManagerDisplayTypeStack:
            _segmentedControl.selectedSegmentIndex = 1;
            break;
    }
    
    _useTopDecorView.on = ([GLBNotificationManager topDecorView] != nil);
    _useBottomDecorView.on = ([GLBNotificationManager bottomDecorView] != nil);
    
    _textField.text = @"Сообщение";
}

- (void)viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];
    
    [_textField becomeFirstResponder];
}

#pragma mark - Actions

- (IBAction)pressedMenu:(id)sender {
    [self.glb_slideViewController showLeftViewControllerAnimated:YES complete:nil];
}

- (IBAction)changeDisplayType:(id)sender {
    [GLBNotificationManager hideAllAnimated:YES complete:^{
        switch(_segmentedControl.selectedSegmentIndex) {
            case 0:
                [GLBNotificationManager setDisplayType:GLBNotificationManagerDisplayTypeList];
                break;
            case 1:
                [GLBNotificationManager setDisplayType:GLBNotificationManagerDisplayTypeStack];
                break;
        }
    }];
}

- (IBAction)changeTopDecorView:(id)sender {
    if(_useTopDecorView.on == YES) {
        [GLBNotificationManager setTopDecorView:[NotificationTopDecorView new]];
    } else {
        [GLBNotificationManager setTopDecorView:nil];
    }
}

- (IBAction)changeBottomDecorView:(id)sender {
    if(_useBottomDecorView.on == YES) {
        [GLBNotificationManager setBottomDecorView:[NotificationBottomDecorView new]];
    } else {
        [GLBNotificationManager setBottomDecorView:nil];
    }
}

- (IBAction)pressedShow:(id)sender {
    NotificationView* view = [NotificationView new];
    view.title = _textField.text;
    [GLBNotificationManager showView:view];
}

@end

File: Demo/Sources/ViewControllers/Notifications/NotificationsViewController.swift
//
//  Globus
//

import Globus

class NotificationsViewController: GLBViewController {
    
    // MARK - Outlet property
    
    @IBOutlet fileprivate weak var segmentedControl: UISegmentedControl!
    @IBOutlet fileprivate weak var useTopDecorView: UISwitch!
    @IBOutlet fileprivate weak var useBottomDecorView: UISwitch!
    @IBOutlet fileprivate weak var textField: GLBTextField!
    
    // MARK - UIViewController
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        GLBNotificationManager.setStatusBarStyle(.lightContent)
        
        self.segmentedControl.removeAllSegments()
        self.segmentedControl.insertSegment(withTitle: "List", at: self.segmentedControl.numberOfSegments, animated: false)
        self.segmentedControl.insertSegment(withTitle: "Stack", at: self.segmentedControl.numberOfSegments, animated: false)
        switch GLBNotificationManager.displayType() {
        case .list:
            self.segmentedControl.selectedSegmentIndex = 0
            break
        case .stack:
            self.segmentedControl.selectedSegmentIndex = 1
            break
        }
        
        self.useTopDecorView.isOn = (GLBNotificationManager.topDecorView() != nil)
        self.useBottomDecorView.isOn = (GLBNotificationManager.bottomDecorView() != nil)
        
        self.textField.text = "Сообщение";
    }
    
    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        
        self.textField.becomeFirstResponder()
    }
    
    // MARK - Action
    
    @IBAction internal func pressedMenu(_ sender: Any) {
        self.glb_slideViewController?.showLeftViewController(animated: true, complete: nil)
    }
    
    @IBAction internal func changeDisplayType(_ sender: Any) {
        switch self.segmentedControl.selectedSegmentIndex {
        case 0:
            GLBNotificationManager.setDisplayType(.list)
            break
        case 1:
            GLBNotificationManager.setDisplayType(.stack)
            break
        default:
            break
        }
    }
    
    @IBAction internal func changeTopDecorView(_ sender: Any) {
        if self.useTopDecorView.isOn {
            GLBNotificationManager.setTopDecorView(NotificationTopDecorView())
        } else {
            GLBNotificationManager.setTopDecorView(nil)
        }
    }
    
    @IBAction internal func changeBottomDecorView(_ sender: Any) {
        if self.useBottomDecorView.isOn {
            GLBNotificationManager.setBottomDecorView(NotificationBottomDecorView())
        } else {
            GLBNotificationManager.setBottomDecorView(nil)
        }
    }
    
    @IBAction internal func pressedShow(_ sender: Any) {
        let view = NotificationView()
        view.title = self.textField.text!
        GLBNotificationManager.show(view)
    }
    
    // MARK: - GLBNibExtension
    
    public override static func nibName() -> String {
        return "NotificationsViewController-Swift"
    }
    
}
